compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/array-to-sorted.js

```diff
@@ -0,0 +1,4 @@
+function Component() {
+  const array = ['c', 'b', 'a'];
+  return useMemo(() => array.toSorted(), [array]);
```

Comment:
For the tests, it isn't sufficient to check that the useeMemo is preserved. We also have to check if the result of calling the function propagates mutability in the correct way. For example, we should add a varaint of this fixture (array-to-sorted-mutate) that mutates the result of toStorted. Something like:

```js
function Component(props) {
  const sorted = useMemo(() => {
    const array = [{value: 'c'}, {value: 'b'}, {value: 'a'}];
    const sorted = array.toSorted();
    mutate(sorted, props.value);
    return sorted;
  }, [props.value]);
  return <ValidateMemoization inputs={[props.value]} output={sorted} />;
}
```